lib/originals-helper.js

```diff
@@ -15,6 +15,29 @@ function isGlobal(node) {
   return false;
 }
 
+/**
+ * @param {Node} node - a ts-morph AST node
+ *
+ * @return {string} the get-originals module name corresponding to the node's
+ * type. (This is sometimes different from the TypeScript type name.)
+ */
+function getModuleNameForNode(node) {
+  return getModuleNameForType(node.getType());
+}
+
+/**
+ * @param {Type} type - a ts-morph type
+ *
+ * @return {string} the get-originals module name corresponding to the
+ * type. (This is sometimes different from the TypeScript type name.)
+ */
+function getModuleNameForType(type) {
+  // `type.getText()` does not work, because it returns things like
+  // "Promise<any>" or "any[]". The name of the type's symbol is what we want:
+  // "Promise" or "Array".
+  return type.getSymbol().getName();
```

Comment:
Is there something more available that lets your distinguish the builtin Array from a random class named Array?

Reply:
Yes; that's done via the isGlobal check that we do at various locations in the code.